module/module.c

```diff
@@ -673,30 +673,28 @@ static int handle_comment_data(merlin_node *node, merlin_header *hdr, void *buf)
 	}
 
 	if (ds->type == NEBTYPE_COMMENT_DELETE) {
-		comment *cmnt, *next_cmnt;
-
-		if (ds->comment_type == HOST_COMMENT) {
-			cmnt = get_first_comment_by_host(ds->host_name);
-			for (; cmnt; cmnt = next_cmnt) {
-				next_cmnt = cmnt->nexthash;
-				if (matching_comment(cmnt, ds)) {
-					merlin_set_block_comment(ds);
-					delete_comment(cmnt->comment_type, cmnt->comment_id);
-					merlin_set_block_comment(NULL);
-				}
-			}
-		} else {
-			/* this is *really* expensive. Sort of wtf? */
-			for (cmnt = comment_list; cmnt; cmnt = next_cmnt) {
-				next_cmnt = cmnt->next;
-
-				if (matching_comment(cmnt, ds)) {
-					merlin_set_block_comment(ds);
-					delete_comment(cmnt->comment_type, cmnt->comment_id);
-					merlin_set_block_comment(NULL);
-				}
+		comment *cmnt;
+		GHashTableIter iter;
+		gpointer comment_;
+
+		if (comment_hashtable == NULL) {
+			ldebug("COMMENTS: Empty comment hashtable. Ignoring delete event");
+			return 0;
+		}
+
+		g_hash_table_iter_init(&iter, comment_hashtable);
+
+		while (g_hash_table_iter_next(&iter, NULL, &comment_)) {
```

Comment:
Though this will probably work, it would be expensive to look through the entire comment list. It seems host and service objects now contain a comment list. kindly check if this can be used instead of the global comment list

Reply:
Updated to use the host/service comment list instead of the global comment list.